### ISSUE TITLE
Map Editor should execute all GUI actions in GUI thread.

### ISSUE THREAD

**Issue by michaelzangl** [jsettlers/settlers-remake#261]
```
Found one Java-level deadlock:
=============================
"XToolkt-Shutdown-Thread":
  waiting for ownable synchronizer 0x00000007b1818438, (a java.util.concurrent.locks.ReentrantLock$NonfairSync),
  which is held by "AWT-EventQueue-0"
"AWT-EventQueue-0":
  waiting to lock monitor 0x00007f86e0004cc8 (object 0x00000007b1558670, a java.awt.Component$AWTTreeLock),
  which is held by "action firerer"
"action firerer":
  waiting for ownable synchronizer 0x00000007b1818438, (a java.util.concurrent.locks.ReentrantLock$NonfairSync),
  which is held by "AWT-EventQueue-0"

Java stack information for the threads listed above:
===================================================
"XToolkt-Shutdown-Thread":
    at sun.misc.Unsafe.park(Native Method)
    - parking to wait for  <0x00000007b1818438> (a java.util.concurrent.locks.ReentrantLock$NonfairSync)
    at java.util.concurrent.locks.LockSupport.park(LockSupport.java:186)
    at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:834)
    at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireQueued(AbstractQueuedSynchronizer.java:867)
    at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(AbstractQueuedSynchronizer.java:1197)
    at java.util.concurrent.locks.ReentrantLock$NonfairSync.lock(ReentrantLock.java:214)
    at java.util.concurrent.locks.ReentrantLock.lock(ReentrantLock.java:290)
    at sun.awt.SunToolkit.awtLock(SunToolkit.java:245)
    at sun.awt.X11.XAtom.<init>(XAtom.java:265)
    at sun.awt.X11.XAtom.<init>(XAtom.java:232)
    at sun.awt.X11.XAtom.get(XAtom.java:178)
    at sun.awt.X11.XSystemTrayPeer.<clinit>(XSystemTrayPeer.java:46)
    at sun.awt.X11.XToolkit$1$1.run(XToolkit.java:242)
"AWT-EventQueue-0":
    at java.awt.Component.getGraphicsConfiguration(Component.java:1146)
    - waiting to lock <0x00000007b1558670> (a java.awt.Component$AWTTreeLock)
    at com.jogamp.opengl.awt.GLCanvas.getGraphicsConfiguration(GLCanvas.java:351)
    at com.jogamp.nativewindow.awt.JAWTWindow.updatePixelScale(JAWTWindow.java:347)
    at com.jogamp.nativewindow.awt.JAWTWindow.updateLockedData(JAWTWindow.java:333)
    at jogamp.nativewindow.jawt.x11.X11JAWTWindow.lockSurfaceImpl(X11JAWTWindow.java:97)
    at com.jogamp.nativewindow.awt.JAWTWindow.lockSurface(JAWTWindow.java:624)
    at jogamp.opengl.GLDrawableImpl.lockSurface(GLDrawableImpl.java:315)
    at jogamp.opengl.GLContextImpl.makeCurrent(GLContextImpl.java:596)
    at jogamp.opengl.GLContextImpl.makeCurrent(GLContextImpl.java:580)
    at jogamp.opengl.GLDrawableHelper.invokeGLImpl(GLDrawableHelper.java:1263)
    at jogamp.opengl.GLDrawableHelper.invokeGL(GLDrawableHelper.java:1131)
    at com.jogamp.opengl.awt.GLCanvas$12.run(GLCanvas.java:1435)
    at com.jogamp.opengl.Threading.invoke(Threading.java:223)
    at com.jogamp.opengl.awt.GLCanvas.display(GLCanvas.java:526)
    at com.jogamp.opengl.awt.GLCanvas.paint(GLCanvas.java:580)
    at com.jogamp.opengl.awt.GLCanvas.update(GLCanvas.java:863)
    at sun.awt.RepaintArea.updateComponent(RepaintArea.java:255)
    at sun.awt.X11.XRepaintArea.updateComponent(XRepaintArea.java:60)
    at sun.awt.RepaintArea.paint(RepaintArea.java:232)
    at sun.awt.X11.XComponentPeer.handleEvent(XComponentPeer.java:592)
    at java.awt.Component.dispatchEventImpl(Component.java:4948)
    at java.awt.Component.dispatchEvent(Component.java:4698)
    at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:747)
    at java.awt.EventQueue.access$300(EventQueue.java:103)
    at java.awt.EventQueue$3.run(EventQueue.java:706)
    at java.awt.EventQueue$3.run(EventQueue.java:704)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
    at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:87)
    at java.awt.EventQueue$4.run(EventQueue.java:720)
    at java.awt.EventQueue$4.run(EventQueue.java:718)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
    at java.awt.EventQueue.dispatchEvent(EventQueue.java:717)
    at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:242)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:161)
    at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:150)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:146)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:138)
    at java.awt.EventDispatchThread.run(EventDispatchThread.java:91)
"action firerer":
    at sun.misc.Unsafe.park(Native Method)
    - parking to wait for  <0x00000007b1818438> (a java.util.concurrent.locks.ReentrantLock$NonfairSync)
    at java.util.concurrent.locks.LockSupport.park(LockSupport.java:186)
    at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:834)
    at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireQueued(AbstractQueuedSynchronizer.java:867)
    at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(AbstractQueuedSynchronizer.java:1197)
    at java.util.concurrent.locks.ReentrantLock$NonfairSync.lock(ReentrantLock.java:214)
    at java.util.concurrent.locks.ReentrantLock.lock(ReentrantLock.java:290)
    at sun.awt.SunToolkit.awtLock(SunToolkit.java:245)
    at sun.awt.X11.XAwtState.getComponentMouseEntered(XAwtState.java:61)
    at sun.awt.X11.XGlobalCursorManager.findHeavyweightUnderCursor(XGlobalCursorManager.java:144)
    at sun.awt.X11.XGlobalCursorManager.findHeavyweightUnderCursor(XGlobalCursorManager.java:160)
    at sun.awt.GlobalCursorManager._updateCursor(GlobalCursorManager.java:179)
    at sun.awt.GlobalCursorManager.updateCursorImmediately(GlobalCursorManager.java:95)
    at sun.awt.X11.XComponentPeer.updateCursorImmediately(XComponentPeer.java:741)
    at java.awt.Component.updateCursorImmediately(Component.java:3131)
    at java.awt.Component.enable(Component.java:1478)
    - locked <0x00000007b1558670> (a java.awt.Component$AWTTreeLock)
    at javax.swing.JComponent.enable(JComponent.java:3610)
    at java.awt.Component.enable(Component.java:1497)
    at java.awt.Component.setEnabled(Component.java:1462)
    at javax.swing.JComponent.setEnabled(JComponent.java:2668)
    at javax.swing.AbstractButton.setEnabled(AbstractButton.java:2087)
    at javax.swing.AbstractAction.setEnabledFromAction(AbstractAction.java:102)
    at javax.swing.AbstractButton.actionPropertyChanged(AbstractButton.java:1216)
    at javax.swing.AbstractButton$ButtonActionPropertyChangeListener.actionPropertyChanged(AbstractButton.java:1361)
    at javax.swing.AbstractButton$ButtonActionPropertyChangeListener.actionPropertyChanged(AbstractButton.java:1350)
    at javax.swing.ActionPropertyChangeListener.propertyChange(ActionPropertyChangeListener.java:88)
    at java.beans.PropertyChangeSupport.fire(PropertyChangeSupport.java:335)
    at java.beans.PropertyChangeSupport.firePropertyChange(PropertyChangeSupport.java:327)
    at javax.swing.event.SwingPropertyChangeSupport.firePropertyChange(SwingPropertyChangeSupport.java:92)
    at java.beans.PropertyChangeSupport.firePropertyChange(PropertyChangeSupport.java:263)
    at javax.swing.AbstractAction.firePropertyChange(AbstractAction.java:276)
    at javax.swing.AbstractAction.setEnabled(AbstractAction.java:236)
    at jsettlers.mapcreator.main.window.EditorFrame.enableAction(EditorFrame.java:185)
    at jsettlers.mapcreator.main.EditorControl.endUseStep(EditorControl.java:741)
    at jsettlers.mapcreator.main.EditorControl.action(EditorControl.java:797)
    at jsettlers.mapcreator.main.EditorControl.fireAction(EditorControl.java:821)
    at jsettlers.mapcreator.main.action.CombiningActionFirerer$ActionFirererThread.run(CombiningActionFirerer.java:63)

Found 1 deadlock
```

**Comment by andreas-eberle** [comment]
is this related to #258? Or is it with the old UI? If it is with the old UI, does it still apear in the better version of #258? If it still exists, let us fix it in #258, as it will replace the old UI.

**Comment by michaelzangl** [comment]
Partially solved in  #272

**Comment by andreas-eberle** [comment]
Is this still an issue?
